Dockerfile

```diff
@@ -98,16 +98,28 @@ ENV CDM_SHARED_DIR=/cdm_shared_workspace
 RUN mkdir -p ${CDM_SHARED_DIR} && chmod -R 777 ${CDM_SHARED_DIR}
 RUN chown -R spark_user:spark $CDM_SHARED_DIR
 
+# Set a directory for hosting Hive metastore files - defined in config/hive-site-template.xml
+ENV HIVE_METASTORE_DIR=$CDM_SHARED_DIR/hive_metastore
+RUN mkdir -p ${HIVE_METASTORE_DIR}
+RUN chown -R spark_user:spark $HIVE_METASTORE_DIR
+
 # Set a directory for hosting Jupyterhub db and cookie secret
 ENV JUPYTERHUB_SECRETS_DIR=/jupyterhub_secrets
 RUN mkdir -p ${JUPYTERHUB_SECRETS_DIR}
 RUN chown -R spark_user:spark ${JUPYTERHUB_SECRETS_DIR}
 
+# Set a directory for hosting postgres db data - used by Spark to store Delta tables
+ENV JUPYTERHUB_POSTGRES_DIR=/cdm-postgres
+RUN mkdir -p ${JUPYTERHUB_POSTGRES_DIR}
+RUN chown -R spark_user:spark ${JUPYTERHUB_POSTGRES_DIR}
+
```

Comment:
I thought we were using a remote postgres instance for hive. What is this for?

Reply:
Yes we are. I am getting confused here too. Do users need to read data from postgres at all? If not, I don't think we need this. I am planning to mount `/cdr/cdm/jupyter/cdm-postgres:/cdm-postgres`. Maybe that's not needed?

Reply:
I assume they need to read / write to the remote postgres as that IIUC, stores the table metadata for the deltalake tables. I don't understand why we'd need local postgres tables at all

Reply:
no. it's not local postgres. But using the remote ones. The remote postgres does `/cdr/cdm/jupyter/cdm-postgres:/var/lib/postgresql/data` in Rancher.

Reply:
If it's remote why does it need a local folder at all?

Reply:
👍 yea. Now I am thinking. You are right. Removed.